### PR TITLE
dbcsr with opencl as a third backend

### PR DESCRIPTION
**Pull request by haampie**
Ping @dev-zero

### CODE REVIEW

File: var/spack/repos/builtin/packages/dbcsr/package.py [hunk 1]
```diff
@@ -60,6 +64,8 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
 
     conflicts('+cuda', when='+rocm', msg="CUDA and ROCm are mutually exclusive")
     conflicts('+rocm', when='@:2.0', msg="ROCm support is available in DBCSR v2.1 and later")
+    conflicts('+cuda', when='+opencl')
+    conflicts('+rocm', when='+opencl')
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/dbcsr/package.py]: conflict needed for version

**Comment by dev-zero** [comment]
breaks older versions of DBCSR

File: var/spack/repos/builtin/packages/dbcsr/package.py [hunk 2]
```diff
@@ -89,8 +95,6 @@ def cmake_args(self):
             '-DLAPACK_LIBRARIES=%s' % (spec['lapack'].libs.joined(';')),
             '-DWITH_EXAMPLES=ON',
             self.define_from_variant('BUILD_SHARED_LIBS', 'shared'),
-            self.define_from_variant('USE_HIP', 'rocm'),
-            self.define_from_variant('USE_CUDA', 'cuda'),
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/dbcsr/package.py:93]: needed for older versions of DBCSR

Comment (dev-zero) [reply var/spack/repos/builtin/packages/dbcsr/package.py:93]: sorry for the blurb, forgot that we (luckily) don't have a version yet for the Spack DBCSR package

File: var/spack/repos/builtin/packages/dbcsr/package.py (hunk 1 above)
Comment (dev-zero) [reply var/spack/repos/builtin/packages/dbcsr/package.py]: forgot that we don't have versions yet in the DBCSR Spack package, sorry!
We can therefore also drop the `when='@:2.0'` from above I made you put in there in the last PR.

**Comment by dev-zero** [comment]
LGTM, thanks!

**Comment by haampie** [comment]
Can you have another look @dev-zero? I've rebased and combined the two prs into this.

File: var/spack/repos/builtin/packages/dbcsr/package.py [hunk 3]
```diff
@@ -59,12 +63,15 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
             conflicts('+rocm', when='amdgpu_target={0}'.format(arch), msg=amd_msg)
 
     conflicts('+cuda', when='+rocm', msg="CUDA and ROCm are mutually exclusive")
-    conflicts('+rocm', when='@:2.0', msg="ROCm support is available in DBCSR v2.1 and later")
+    conflicts('+cuda', when='+opencl')
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/dbcsr/package.py]: Can you add a `msg` that `opencl` and `cuda` are mutually exclusive?

Comment (haampie) [reply var/spack/repos/builtin/packages/dbcsr/package.py]: Yes

**Comment by dev-zero** [comment]
Still good :)

**Review by alalazo — APPROVED** [review]

**Pull request merged**